### PR TITLE
add ZH translation /docs/tasks/federation/federation-service-discover…

### PR DESCRIPTION
**Pull request by tanjunchen**

### CODE REVIEW

File: content/zh/docs/tasks/federation/federation-service-discovery.md [hunk 1]
```diff
@@ -0,0 +1,508 @@
+---
+title: 使用联合服务的跨集群服务发现
```
Comment (chenrui333) [inline content/zh/docs/tasks/federation/federation-service-discovery.md]:
```suggestion
title: 使用联合服务来实现跨集群服务发现
```

File: content/zh/docs/tasks/federation/federation-service-discovery.md [hunk 2]
```diff
@@ -0,0 +1,508 @@
...
+
+{{< deprecationfilewarning >}}
+{{< include "federation-deprecation-warning-note.md" >}}
+{{< /deprecationfilewarning >}}
+
+<!-- This guide explains how to use Kubernetes Federated Services to deploy
+a common Service across multiple Kubernetes clusters. This makes it
+easy to achieve cross-cluster service discovery and availability zone
+fault tolerance for your Kubernetes applications. -->
+本指南说明了如何使用 Kubernetes 联合服务跨多个 Kubernetes 集群部署通用服务。这样可以轻松实现 Kubernetes 应用程序的跨集群服务发现和可用性区域容错。
```
Comment (chenrui333) [inline content/zh/docs/tasks/federation/federation-service-discovery.md]: AZ avail zone, 直接翻译成 `可用区` 就行了

Comment (chenrui333) [inline content/zh/docs/tasks/federation/federation-service-discovery.md]:
```suggestion
本指南说明了如何使用 Kubernetes 联合服务跨多个 Kubernetes 集群部署通用服务。这样可以轻松实现 Kubernetes 应用程序的跨集群服务发现和可用区容错。
```

File: content/zh/docs/tasks/federation/federation-service-discovery.md [hunk 3]
```diff
@@ -0,0 +1,508 @@
...
+
+
+<!-- Federated Services are created in much that same way as traditional
+[Kubernetes Services](/docs/concepts/services-networking/service/) by making an API
+call which specifies the desired properties of your service. In the
+case of Federated Services, this API call is directed to the
+Federation API endpoint, rather than a Kubernetes cluster API
+endpoint. The API for Federated Services is 100% compatible with the
+API for traditional Kubernetes Services. -->
+联合服务的创建与传统服务几乎相同[Kubernetes Services](/docs/concepts/services-networking/service/)即通过 API 调用来指定所需的服务属性。对于联合服务，此 API 调用定向到联合身份验证 API 接入点，而不是 Kubernetes 集群 API 接入点。联合服务的 API 与传统 Kubernetes 服务的 API 是100%兼容的。
```
Comment (chenrui333) [inline content/zh/docs/tasks/federation/federation-service-discovery.md]:
```suggestion
联合服务的创建与传统服务几乎相同 [Kubernetes Services](/docs/concepts/services-networking/service/) 即通过 API 调用来指定所需的服务属性。对于联合服务，此 API 调用定向到联合身份验证 API 接入点，而不是 Kubernetes 集群 API 接入点。联合服务的 API 与传统 Kubernetes 服务的 API 是100%兼容的。
```

File: content/zh/docs/tasks/federation/federation-service-discovery.md [hunk 4]
```diff
@@ -0,0 +1,508 @@
...
+<!-- Once created, the Federated Service automatically: -->
+创建后，联合服务会自动:
+
+<!-- 1. Creates matching Kubernetes Services in every cluster underlying your Cluster Federation,
+2. Monitors the health of those service "shards" (and the clusters in which they reside), and
+3. Manages a set of DNS records in a public DNS provider (like Google Cloud DNS, or AWS Route 53), thus ensuring that clients
+of your federated service can seamlessly locate an appropriate healthy service endpoint at all times, even in the event of cluster,
+availability zone or regional outages. -->
+1. 在基础集群联合的每个集群中创建匹配的 Kubernetes 服务,
+2. 监视那些服务 "分片"(及其驻留的集群)的运行状况, and
```
Comment (chenrui333) [inline content/zh/docs/tasks/federation/federation-service-discovery.md]:
```suggestion
2. 监视那些服务 "分片"(及其驻留的集群)的运行状况, 以及
```

File: content/zh/docs/tasks/federation/federation-service-discovery.md [hunk 5]
```diff
@@ -0,0 +1,508 @@
...
+3. 在公共 DNS 提供商(例如 Google Cloud DNS 或 AWS Route 53)中管理一组 DNS 记录,即使在集群可用区域中断的情况下，也能确保您联合服务的客户端始终可以无缝地定位合适的健康服务接入点。
+
+<!-- Clients inside your federated Kubernetes clusters (that is Pods) will
+automatically find the local shard of the Federated Service in their
+cluster if it exists and is healthy, or the closest healthy shard in a
+different cluster if it does not. -->
+如果存在健康的分片，联合 Kubernetes 集群(即Pods)中的客户端将自动在其中找到联合服务的本地分片集群或者集群中最接近的健康分片;如果不存在健康的分片，则使用其他群集。
```
Comment (chenrui333) [inline content/zh/docs/tasks/federation/federation-service-discovery.md]:
```suggestion
如果存在健康的分片，联合 Kubernetes 集群(即 Pods )中的客户端将自动在其中找到联合服务的本地分片集群或者集群中最接近的健康分片；如果不存在健康的分片，则使用其他集群。
```

File: content/zh/docs/tasks/federation/federation-service-discovery.md [hunk 6]
```diff
@@ -0,0 +1,508 @@
...
+## 前提
+
+<!-- This guide assumes that you have a running Kubernetes Cluster
+Federation installation. If not, then head over to the
+[federation admin guide](/docs/admin/federation/) to learn how to
+bring up a cluster federation (or have your cluster administrator do
+this for you). Other tutorials, for example
+[this one](https://github.com/kelseyhightower/kubernetes-cluster-federation)
+by Kelsey Hightower, are also available to help you. -->
+本指南假设您已经安装 Kubernetes 联合集群。如果没有，则访问[federation admin guide](/docs/admin/federation/)了解如何建立联合集群(或让您的集群管理员为您执行此操作)。其他教程，例如 Kelsey Hightower 编写的[案例](https://github.com/kelseyhightower/kubernetes-cluster-federation)或许有用。
```
Comment (chenrui333) [inline content/zh/docs/tasks/federation/federation-service-discovery.md]:
```suggestion
本指南假设您已经安装 Kubernetes 联合集群。如果没有，则访问[联合集群管理指南](/docs/admin/federation/)了解如何建立联合集群(或让您的集群管理员为您执行此操作)。其他教程，例如 Kelsey Hightower 编写的[案例](https://github.com/kelseyhightower/kubernetes-cluster-federation)或许有用。
```

File: content/zh/docs/tasks/federation/federation-service-discovery.md [hunk 7]
```diff
@@ -0,0 +1,508 @@
...
+
+<!-- You should also have a basic
+[working knowledge of Kubernetes](/docs/tutorials/kubernetes-basics/) in
+general, and [Services](/docs/concepts/services-networking/service/) in particular. -->
+一般而言，您应该还有基本的[Kubernetes 工作常识](/docs/tutorials/kubernetes-basics/)，特别是[Services](/docs/concepts/services-networking/service/)。
```
Comment (chenrui333) [inline content/zh/docs/tasks/federation/federation-service-discovery.md]:
```suggestion
一般而言，您应该还有基本的 [Kubernetes 工作常识](/docs/tutorials/kubernetes-basics/)，特别是 [Services](/docs/concepts/services-networking/service/)。
```

File: content/zh/docs/tasks/federation/federation-service-discovery.md [hunk 8]
```diff
@@ -0,0 +1,508 @@
...
+aws route53 list-hosted-zones
+```
+and
+
+``` shell
+aws route53 list-resource-record-sets --hosted-zone-id Z3ECL0L9QLOVBX
+```
+{{< /note >}} -->
+{{< note >}}
+如果您的联邦配置为使用 AWS Route53，则可以使用等效的 AWS 工具，例如:
```
Comment (chenrui333) [inline content/zh/docs/tasks/federation/federation-service-discovery.md]:
```suggestion
如果您的联邦配置为使用 AWS Route53，则可以使用类似的 AWS 工具，例如:
```

File: content/zh/docs/tasks/federation/federation-service-discovery.md [hunk 9]
```diff
@@ -0,0 +1,508 @@
...
+{{< /note >}}
+
+<!-- Whatever DNS provider you use, any DNS query tool (for example 'dig'
+or 'nslookup') will of course also allow you to see the records
+created by the Federation for you. Note that you should either point
+these tools directly at your DNS provider (such as `dig
+@ns-cloud-e1.googledomains.com...`) or expect delays in the order of
+your configured TTL (180 seconds, by default) before seeing updates,
+due to caching by intermediate DNS servers. -->
+无论使用哪种 DNS 提供程序，任何 DNS 查询工具(例如 'dig' 或者 'nslookup')当然都将允许您查看联邦会为您创建的记录。请注意，您应该将这些工具直接指向您的 DNS 提供商(例如 `dig @ ns-cloud-e1.googledomains.com ...`),或者由于中间 DNS 服务器进行了缓存，因此在看到更新之前，预计延迟会按照配置的 TTL 顺序(默认为180秒)进行。
```
Comment (chenrui333) [inline content/zh/docs/tasks/federation/federation-service-discovery.md]:
```suggestion
无论使用哪种 DNS 提供商，任何 DNS 查询工具(例如 'dig' 或者 'nslookup')都将允许您查看联邦会为您创建的记录。请注意，您应该将这些工具直接指向您的 DNS 提供商(例如 `dig @ ns-cloud-e1.googledomains.com ...`),或者由于中间 DNS 服务器进行了缓存，因此在看到更新之前，预计延迟会按照配置的 TTL 顺序(默认为 180 秒)进行。
```

File: content/zh/docs/tasks/federation/federation-service-discovery.md [hunk 10]
```diff
@@ -0,0 +1,508 @@
...
+3. For zones where there are currently no healthy backend endpoints, a CNAME ('Canonical Name') record is used to alias (automatically redirect) those queries to the next closest healthy zone.  In the example, the service shard in us-central1-f currently has no healthy backend endpoints (that is Pods), so a CNAME record has been created to automatically redirect queries to other shards in that region (us-central1 in this case).
+4. Similarly, if no healthy shards exist in the enclosing region, the search progresses further afield. In the europe-west1-d availability zone, there are no healthy backends, so queries are redirected to the broader europe-west1 region (which also has no healthy backends), and onward to the global set of healthy addresses (' nginx.mynamespace.myfederation.svc.example.com.'). -->
+1. 请注意，每个具有至少一个正常后端端点的服务分片都有一条正常('A')记录。例如，在 us-central1-a 中，104.197.247.191是该区域中服务分片的外部 IP 地址，在 asia-east1-a 中，该地址是 130.211.56.221。
```
Comment (chenrui333) [inline content/zh/docs/tasks/federation/federation-service-discovery.md]:
```suggestion
1. 请注意，每个具有至少一个正常后端端点的服务分片都有一条正常('A')记录。例如，在 us-central1-a 中，104.197.247.191 是该区域中服务分片的外部 IP 地址，在 asia-east1-a 中，该地址是 130.211.56.221。
```

File: content/zh/docs/tasks/federation/federation-service-discovery.md [hunk 11]
```diff
@@ -0,0 +1,508 @@
...
+then select any one of the returned addresses to initiate a network
+connection (and fail over automatically to one of the other equivalent
+addresses if required). -->
+上面的 DNS 记录集由联邦服务系统自动与全球所有服务分片的当前健康状况保持同步。DNS 解析库(由所有客户端调用)自动遍历 'CNAME' 与 'A' 记录的层次结构，以返回正确健康的 IP 地址集。然后，客户端可以选择任何返回的地址来启动网络连接(并根据需要自动故障转移到其他等效地址之一)。
+
+<!-- ## Discovering a federated service -->
+## 发现联合服务
+
+<!-- ### From pods inside your federated clusters -->
+### 从联合集群内的 Pods
```
Comment (chenrui333) [inline content/zh/docs/tasks/federation/federation-service-discovery.md]:
```suggestion
### 从联合集群内的 Pods 来发现
```

File: content/zh/docs/tasks/federation/federation-service-discovery.md [hunk 12]
```diff
@@ -0,0 +1,508 @@
...
+
+<!-- By default, Kubernetes clusters come pre-configured with a
+cluster-local DNS server ('KubeDNS'), as well as an intelligently
+constructed DNS search path which together ensure that DNS queries
+like "myservice", "myservice.mynamespace",
+"bobsservice.othernamespace" etc issued by your software running
+inside Pods are automatically expanded and resolved correctly to the
+appropriate service IP of services running in the local cluster. -->
+默认情况下，Kubernetes 集群预先配置了本地集群 DNS 服务器('KubeDNS')以及智能构建的 DNS 搜索路径，这些路径共同确保由 Pods 内部运行软件发出的 DNS 查询如"myservice", "myservice.mynamespace","bobsservice.othernamespace"等，会自动扩展并正确解析为本地集群运行服务的相应服务 IP。
```
Comment (chenrui333) [inline content/zh/docs/tasks/federation/federation-service-discovery.md]:
```suggestion
默认情况下，Kubernetes 集群预先配置了本地集群 DNS 服务器('KubeDNS')以及智能构建的 DNS 搜索路径，这些路径共同确保由 Pods 内部运行软件发出的 DNS 查询如 "myservice", "myservice.mynamespace","bobsservice.othernamespace" 等，会自动扩展并正确解析为本地集群运行服务的相应服务 IP。
```

File: content/zh/docs/tasks/federation/federation-service-discovery.md [hunk 13]
```diff
@@ -0,0 +1,508 @@
...
+running in any other cluster across your Cluster Federation, globally.
+To take advantage of this extended range, you use a slightly different
+DNS name of the form ```"<servicename>.<namespace>.<federationname>"```
+to resolve Federated Services. For example, you might use
+`myservice.mynamespace.myfederation`. Using a different DNS name also
+avoids having your existing applications accidentally traversing
+cross-zone or cross-region networks and you incurring perhaps unwanted
+network charges or latency, without you explicitly opting in to this
+behavior. -->
+随着联合服务和跨集群服务发现的引入，该概念已扩展到涵盖在全球集群联盟中任何其他集群中运行的 Kubernetes 服务。要利用此扩展范围，您可以使用形式稍有不同的 DNS 名称，形式为 ```"<servicename>.<namespace>.<federationname>"```来解析联合服务。例如，您可以使用 `myservice.mynamespace.myfederation`。使用不同的 DNS 名称还可以避免现有应用程序意外穿越跨区域或跨区域网络，并且可能招致不必要的网络费用或延迟，而无需您明确选择采取这种行为。
```
Comment (chenrui333) [inline content/zh/docs/tasks/federation/federation-service-discovery.md]:
```suggestion
随着联合服务和跨集群服务发现的引入，该概念已扩展到涵盖在全球集群联盟中任何其他集群中运行的 Kubernetes 服务。要利用此扩展范围，您可以使用形式稍有不同的 DNS 名称，形式为 ```"<servicename>.<namespace>.<federationname>"``` 来解析联合服务。例如，您可以使用 `myservice.mynamespace.myfederation`。使用不同的 DNS 名称还可以避免现有应用程序意外穿越跨区域或跨区域网络，并且可能招致不必要的网络费用或延迟，而无需您明确选择采取这种行为。
```

File: content/zh/docs/tasks/federation/federation-service-discovery.md [hunk 14]
```diff
@@ -0,0 +1,508 @@
...
+but has no healthy backend pods), the DNS query is automatically
+expanded to ```"nginx.mynamespace.myfederation.svc.us-central1-f.example.com"```
+(that is, logically "find the external IP of one of the shards closest to
+my availability zone"). This expansion is performed automatically by
+KubeDNS, which returns the associated CNAME record. This results in
+automatic traversal of the hierarchy of DNS records in the above
+example, and ends up at one of the external IPs of the Federated
+Service in the local us-central1 region (that is 104.197.247.191,
+104.197.244.180 or 104.197.245.170). -->
+但是，如果服务在本地集群中不存在(或者存在但没有正常的后端 Pod),则 DNS 查询会自动扩展为 ```"nginx.mynamespace.myfederation.svc.us-central1-f.example.com"```(也就是说，从逻辑上 "找到最接近我可用区的一个分片的外部 IP")。此扩展由 KubeDNS 自动执行，它返回关联的 CNAME 记录。这将导致在上面的示例中自动遍历 DNS 记录的层次结构，并最终到达本地 us-central1 区域中联合服务的外部 IP 之一(即104.197.247.191,104.197.244.180或104.197.245.170)。
```
Comment (chenrui333) [inline content/zh/docs/tasks/federation/federation-service-discovery.md]:
```suggestion
但是，如果服务在本地集群中不存在(或者存在但没有正常的后端 Pod),则 DNS 查询会自动扩展为 ```"nginx.mynamespace.myfederation.svc.us-central1-f.example.com"```(也就是说，从逻辑上 "找到最接近我可用区的一个分片的外部 IP")。此扩展由 KubeDNS 自动执行，它返回关联的 CNAME 记录。这将导致在上面的示例中自动遍历 DNS 记录的层次结构，并最终到达本地 us-central1 区域中联合服务的外部 IP 之一(即 104.197.247.191, 104.197.244.180 或 104.197.245.170 )。
```

File: content/zh/docs/tasks/federation/federation-service-discovery.md [hunk 15]
```diff
@@ -0,0 +1,508 @@
...
+检查您的
+
+<!-- 1. Client (typically kubectl) is correctly configured (including API endpoints and login credentials).
+2. Cluster Federation API server is running and network-reachable. -->
+1. 客户端(通常是 kubectl)已正确配置(包括 API 端点和登录凭据)。
+2. 联合集群 API 服务器正在运行并且可以访问网络。
+
+<!-- See the [federation admin guide](/docs/admin/federation/) to learn
+how to bring up a cluster federation correctly (or have your cluster administrator do this for you), and how to correctly configure your client. -->
+请参阅[联合管理员指南](/docs/admin/federation/)了解如何正确启动联邦集群(或让您的集群管理员为您执行此操作),以及如何正确配置客户端。
```
Comment (chenrui333) [inline content/zh/docs/tasks/federation/federation-service-discovery.md]:
```suggestion
请参阅[联合集群管理员指南](/docs/admin/federation/)了解如何正确启动联邦集群(或让您的集群管理员为您执行此操作),以及如何正确配置客户端。
```

File: content/zh/docs/tasks/federation/federation-service-discovery.md [hunk 16]
```diff
@@ -0,0 +1,508 @@
...
+Check that: -->
+### 我可以成功创建联合服务，但是在我的 DNS 提供程序中没有创建匹配的 DNS 记录。
+检查:
+
+<!-- 1. Your federation name, DNS provider, DNS domain name are configured correctly.  Consult the [federation admin guide](/docs/admin/federation/) or  [tutorial](https://github.com/kelseyhightower/kubernetes-cluster-federation) to learn
+how to configure your Cluster Federation system's DNS provider (or have your cluster administrator do this for you).
+2. Confirm that the Cluster Federation's service-controller is successfully connecting to and authenticating against your selected DNS provider (look for `service-controller` errors or successes in the output of `kubectl logs federation-controller-manager --namespace federation`).
+3. Confirm that the Cluster Federation's service-controller is successfully creating DNS records in your DNS provider (or outputting errors in its logs explaining in more detail what's failing). -->
+1. 您的联邦名称，DNS 提供程序，DNS 域名已正确配置。请参阅[联邦管理指南](/docs/admin/federation/)或者[教程](https://github.com/kelseyhightower/kubernetes-cluster-federation)了解如何配置联合集群系统的 DNS 提供程序(或让您的集群管理员为您执行此操作)。
```
Comment (chenrui333) [inline content/zh/docs/tasks/federation/federation-service-discovery.md]:
```suggestion
1. 您的联邦集群名称，DNS 提供程序，DNS 域名已正确配置。请参阅[联邦集群管理指南](/docs/admin/federation/)或者[教程](https://github.com/kelseyhightower/kubernetes-cluster-federation)了解如何配置联合集群系统的 DNS 提供程序(或让您的集群管理员为您执行此操作)。
```

File: content/zh/docs/tasks/federation/federation-service-discovery.md [hunk 17]
```diff
@@ -0,0 +1,508 @@
...
+2. 确认联合集群的服务控制器已成功连接到所选的 DNS 提供程序并对其进行身份验证(在 `kubectl logs federation-controller-manager --namespace federation` 的输出中查找 `service-controller` 错误或者成功)。
+3. 确认联合集群的服务控制器已在您的 DNS 提供程序中成功创建了 DNS 记录(或在其日志中输出错误，以更详细地解释失败原因)。
+
+<!-- ### Matching DNS records are created in my DNS provider, but clients are unable to resolve against those names
+Check that: -->
+### 在我的 DNS 提供程序中创建了匹配的 DNS 记录，但是客户端无法根据这些名称进行解析
+检查:
+
+<!-- 1. The DNS registrar that manages your federation DNS domain has been correctly configured to point to your configured DNS provider's nameservers.  See for example [Google Domains Documentation](https://support.google.com/domains/answer/3290309?hl=en&ref_topic=3251230) and [Google Cloud DNS Documentation](https://cloud.google.com/dns/update-name-servers), or equivalent guidance from your domain registrar and DNS provider. -->
+1. 已正确配置用于管理联合 DNS 域名的 DNS 注册器，使其指向已配置的 DNS 提供程序的名称服务器。例如，请参见[Google Domains Documentation](https://support.google.com/domains/answer/3290309?hl=en&ref_topic=3251230)与[Google Cloud DNS Documentation](https://cloud.google.com/dns/update-name-servers),或者域名注册商和 DNS 提供商的等效指南。
```
Comment (chenrui333) [inline content/zh/docs/tasks/federation/federation-service-discovery.md]:
```suggestion
1. 已正确配置用于管理联合 DNS 域名的 DNS 注册器，使其指向已配置的 DNS 提供程序的名称服务器。例如，请参见[Google Domains 文档](https://support.google.com/domains/answer/3290309?hl=en&ref_topic=3251230)与[Google Cloud DNS 文档](https://cloud.google.com/dns/update-name-servers),或者域名注册商和 DNS 提供商的等效指南。
```

**Comment by tanjunchen** [comment]
@chenrui333  Thank you very much for your comments . I will  take  it.

**Comment by tanjunchen** [comment]
@chenrui333  Can you help me review   [translation-zh](https://github.com/kubernetes/website/pull/16672)? Thanks!!!

**Comment by chenrui333** [comment]
@tanjunchen 谢谢你的贡献！！

**Pull request merged**